EXTENSIONS.md

```diff
@@ -0,0 +1,236 @@
+
+## Annotation syntax
+
+**Annotations**, a declarative complement to decorators, use the syntax `@{ }`, which behaves exactly like an object literal. Arbitrary expressions, spread, computed property names, etc, are permitted. It is accessible from the annotated object through the `[Symbol.metadata]` property.
+
+```mjs
+@{x: "y"} @{v: "w"} class C {
+  @{a: "b"} method() { }
+  @{c: "d"} field;
+}
+
+C[Symbol.metadata].class.x                     // "y"
+C[Symbol.metadata].class.v                     // "w"
+C[Symbol.metadata].prototype.methods.method.a  // "a"
+C[Symbol.metadata].instance.fields.field.c     // "d"
+```
+
+Annotations must always have a literal `@{` to start them. To use an existing object as an annotation, you can use the syntax `@{ ...obj }`. The entirety of object syntax is available, including computed property names, arbitrary expressions as values, shorthand names, concise methods, etc.
+
+Libraries and frameworks which want to establish consistent conventions for using annotations may do so based on a Symbol property key that they export. Annotations have the potential advantage in load time performance that engines can directly execute them, as they are as declarative as an object literal.
+
+Annotation semantics may be useful for cases like ORMs and serialization frameworks, which need information about class fields, without affecting their normal runtime semantics. However, the popular ecosystem examples that we've found of this form, using just metadata for fields, seem to depend on metadata generated by TypeScript types. From these examples, it seems that annotation syntax alone would not be a sufficient solution.
+
+Some frameworks, including Angular, tend to use decorators which operate primarily by adding metadata. However, object literal annotations are not quite suitable for this usage, as they don't provide a way to be annotated in TypeScript to check types the way that functions do, they don't allow any processing of the metadata in code before it is saved, and because they don't provide a usable, stable identifier to be used for custom static analysis tools like tree shaking. For Angular, it may make more sense to use decorators which add metadata.
+
+For these reasons, annotations are omitted from this proposal's "MVP" (minimum viable product) and considered as a possible follow-on proposal.
+
+## Function decorators and annotations
+
+The `@logged` decorator from earlier would Just Work(TM) on a function, with function decorators!
+
+```js
+@logged
+@{x: "y"}
+function f() { }
+
+f();                        // prints logging information
+f[Symbol.annotations][0].x  // "y"
```

Comment:
What's the difference between Symbol.metadata and Symbol.annotations? Does it require to be 2 new symbols instead of 1?

Reply:
No difference; I meant to switch everything to metadata and left out some cases ...

Reply:
No difference; I meant to switch everything to metadata and left out some cases ...